### PR TITLE
Make sim driver classes' time variables private

### PR DESCRIPTION
**Pull request by patkenneally**
* **Tickets addressed:** bsk-000
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description
This PR prepares the way to replace the current representation of time within the system as a `uint64_t` with a duration/instant type which defines safe operations to manipulate the duration and request it's value as different units (usec, msec, sec, min, etc.). The PR adds in requisite Swig generated deprecation warnings for the benefit of the third party users of this repository. However, these deprecation warnings are unlikely to see their own removal as we've begun work to remove Swig from the project. 

The PR carries out each of the following steps on each simulation driver class: add accessor methods, add deprecation warning, and change the time variable's scopes to private. The `SimThreadExecution` class is an exception as it's an
implementation detail which doesn't affect the sim driver's interface.

## Verification
All tests pass. Further, at each commit where the deprecation warnings are added, one can verify that the warnings are thrown while normal operation continues.

## Documentation
This does invalidate some documentation. This documentation will be updated when the follow on PR (implementing the `Duration` type) is completed.

## Future work
Follow on `Duration` type PR.

### CODE REVIEW

File: src/architecture/system_model/sim_model.cpp [hunk 1]
```diff
@@ -307,7 +335,7 @@ void SimModel::StepUntilStop(uint64_t SimStopTime, int64_t stopPri)
     }
     for(thrIt=this->threadList.begin(); thrIt != this->threadList.end(); thrIt++)
```
Comment (brsm3129) [inline src/architecture/system_model/sim_model.cpp:336]: I had to go back up and read the definition of the for loop to figure out what thrIt was, could be named something more clear like thread or curThread but this is a minor change and this name is not new in this PR so not a blocking issue.

**Review by brsm3129 — APPROVED** [review]
Looks clean and thorough. I left a comment about an opportunity to improve readability but not a pressing change.

File: src/architecture/system_model/sim_model.cpp (hunk 1 above)
Comment (patkenneally) [reply src/architecture/system_model/sim_model.cpp:336]: Good call and agreed. I change all these in the follow up PR https://github.com/lasp/basilisk/pull/210

**Review by thibaudteil — APPROVED** [review]
Nice!

**Pull request merged**